### DISCUSSION TITLE
How to export extended Prisma Client? ❔

### DISCUSSION THREAD

**Discussion by meotimdihia** [prisma/prisma#19930] [Q&A] [answered]
How to export extended Prisma Client? I am getting the error: 

```
Exported variable 'prisma' has or is using name 'PrismaPromise_2' from external module "/Users/meotimdihia/Sources//node_modules/@prisma/client/runtime/index" but cannot be named.
```
```ts
import { PrismaClient, Prisma } from "@prisma/client"
i

// PrismaClient is attached to the `global` object in development to prevent
// exhausting your database connection limit.
//
// Learn more:
// https://pris.ly/d/help/next-js-best-practices

interface Global {
  prisma: PrismaClient | undefined
}
declare const global: Global

const prisma =
  global.prisma ||
  new PrismaClient().$extends({
    query: {
      users: {
        async $allOperations({ args, query, operation, model }) {
          return query(args)
        }
      }
    }
  })

export default prisma
```

![image](https://github.com/prisma/prisma/assets/300961/ce576011-5888-4fb1-8857-7cb5852e0665)

**Comment by millsp** [comment, marked as answer]
Hey, which version of Prisma Client are you using? I think this was broken in `4.16.0` but fixed in `4.16.1`.

**Reply by meotimdihia** [reply]
Thanks, I just upgraded Prisma to 4.16.1 but have got a new error:

`Type 'never' has no call signatures`.  

Edit: It looks like the bug was reported in the issue: https://github.com/prisma/prisma/issues/19888  
   
![image](https://github.com/prisma/prisma/assets/300961/29cecce0-78e2-4f24-a547-fc482f3ed034)
![image](https://github.com/prisma/prisma/assets/300961/e68c6295-fa5d-4101-baba-d7314e832429)
I don't get this problem with 4.16.0

**Reply by millsp** [reply]
Does upgrading to `4.16.2-dev.2` fix it for you?

**Reply by meotimdihia** [reply]
![image](https://github.com/prisma/prisma/assets/300961/5ce795b1-134b-46ad-84f9-6ff2fe162cae)


```ts
import { PrismaClient, Prisma } from "@prisma/client"

// PrismaClient is attached to the `global` object in development to prevent
// exhausting your database connection limit.
//
// Learn more:
// https://pris.ly/d/help/next-js-best-practices

interface Global {
  prisma: PrismaClient | undefined
}
declare const global: Global

const prisma = global.prisma || new PrismaClient().$extends({
  query: {
    $allModels: {
      $allOperations({ model, operation, args, query }) {
        /* your custom logic for modirying all operations on all models here */
        return query(args)
      }
    }
  }
})

export default prisma

```
I upgrade to 4.16.2-dev.2. The problem is still not fixed :)

**Reply by millsp** [reply]
I cannot reproduce, could you please share your schema?

**Reply by meotimdihia** [reply]
@millsp I upgraded to `4.17.0-dev.37` and it fixed the problem. 👍 Thanks for following the question.

**Reply by millsp** [reply]
You're welcome, thanks for the report.
